plugin/pkg/scheduler/algorithm/priorities/util/topologies.go

```diff
@@ -17,34 +17,57 @@ limitations under the License.
 package util
 
 import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/util/sets"
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
 	"k8s.io/kubernetes/pkg/api/v1"
 	"k8s.io/kubernetes/pkg/features"
 )
 
+// NamespaceLister helps list Namespaces.
+type namespaceLister interface {
+	// List lists all Namespaces in the indexer.
+	List(selector labels.Selector) (ret []*v1.Namespace, err error)
+}
+
 // GetNamespacesFromPodAffinityTerm returns a set of names
-// according to the namespaces indicated in podAffinityTerm.
-// If namespaces is empty it considers the given pod's namespace.
-func GetNamespacesFromPodAffinityTerm(pod *v1.Pod, podAffinityTerm *v1.PodAffinityTerm) sets.String {
-	names := sets.String{}
-	if len(podAffinityTerm.Namespaces) == 0 {
-		names.Insert(pod.Namespace)
-	} else {
-		names.Insert(podAffinityTerm.Namespaces...)
+// according to the NamespaceSelector indicated in podAffinityTerm.
+// If NamespaceSelector is not provided, podAffinityTerm.Namespaces will be used and if it is empty,
+// pod's namespace will be returned.
+// TODO We will remove the filed term.Namespaces in 1.8, then we need to remove using the term.Namespaces here
+func GetNamespacesFromPodAffinityTerm(nsLister namespaceLister, pod *v1.Pod, term *v1.PodAffinityTerm) sets.String {
+	namespaces := sets.String{}
+	namespaceSelector, err := metav1.LabelSelectorAsSelector(term.NamespaceSelector)
```

Comment:
@wojtek-t If namespaceSelector is not set, means nil, `metav1.LabelSelectorAsSelector(term.NamespaceSelector)` will return `labels.Nothing()`, so nsLister.List(namespaceSelector) will return a empty list of namespaces, if so, we will use the `namespaces` field instead. and also we have validation code to make sure they cannot coexists.